modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -88,6 +90,107 @@ TEST_P(BuildPyr_CalcOptFlow_PipelineTest, AccuracyTest)
+    // Preserving 'in' in output to have possibility to compare with OpenCV reference
+    cv::GComputation c(cv::GIn(in), cv::GOut(cv::gapi::copy(in), out));
+
+    // G-API compilation of graph for streaming mode
+    auto gapiBackSub = c.compileStreaming(getCompileArgs());
+
+    // Testing G-API Background Substractor in streaming mode
+    try
+    {
+        gapiBackSub.setSource(gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+                                                      (findDataFile(filePath)));
+    }
+    catch (...)
+    { throw SkipTestException("Video file can't be opened."); }
+
+    cv::Ptr<cv::BackgroundSubtractor> pOCVBackSub;
+
+    if (opType == cv::gapi::video::TYPE_BS_MOG2)
+        pOCVBackSub = cv::createBackgroundSubtractorMOG2(histLength, thr,
+                                                         detectShadows);
+    else if (opType == cv::gapi::video::TYPE_BS_KNN)
+        pOCVBackSub = cv::createBackgroundSubtractorKNN(histLength, thr,
+                                                        detectShadows);
+
+    // Allowing 1% difference of all pixels between G-API and reference OpenCV results
+    testBackgroundSubtractorStreaming(gapiBackSub, pOCVBackSub, 1, 1, learningRate, testNumFrames);
+}
+
+TEST_P(KalmanFilterTest, AccuracyTest)
+{
+    GAPI_Assert(depth == CV_32F || depth == CV_64F);
+    GAPI_Assert(dDim > 0 && mDim > 0 && cDim >= 0);
+
+    cv::gapi::video::KalmanParams kp(dDim, mDim, cDim, depth);
+
+    //measurements vector
+    cv::Mat measure_vec(mDim, 1, depth);
+
+    //control vector
+    cv::Mat ctrl_vec;
+
+    if (cDim > 0)
+        ctrl_vec = Mat::zeros(cDim, 1, depth);
+    else if (cDim == 0)
+        ctrl_vec = Mat::zeros(dDim, 1, depth);
+
+    cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
+
+    if (cDim > 0)
+        cv::randu(ctrl_vec, Scalar::all(-1), Scalar::all(1));
+
+    cv::Mat gapiKState(dDim, 1, depth);
+    cv::Mat ocvKState(dDim, 1, depth);
+
+    // G-API code
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation comp(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+
+    cv::Mat gapiKalmanOut;
+    //std::vector<cv::Mat> mats = { measure_vec, ctrl_vec};
+    comp.apply(cv::gin(measure_vec, haveMeasure, ctrl_vec), cv::gout(gapiKalmanOut));
+    //auto gapiKalmanout = c.compile(cv::descrs_of(mats),getCompileArgs());
+
+    //gapiKalman(cv::gin(measure_vec, ctrl_vec), cv::gout(gapiKState));
+
+    cv::KalmanFilter ocvKalman(dDim, mDim, cDim, depth);
+
+    if (cDim > 0)
+        ocvKalman.predict(ctrl_vec);
+    else if (cDim == 0)
+        ocvKalman.predict();
+
+    ocvKState = ocvKalman.correct(measure_vec);
+
+    // Comparison //////////////////////////////////////////////////////////////
+    {
+        double diff = 0;
+        vector<int> idx;
+        const double eps = 1.0;
+        EXPECT_TRUE(cmpEps(gapiKState, ocvKState, &diff, eps, &idx, false) >= 0);
```

Comment:
Why don't give these values in-place? Doesn't it work?

Reply:
Because `cmpEps()` requires **pointers** to `vector<int>` and to `double diff`, so we have to create this objects to have possibility to pass pointers to them.

Reply:
Oh, I see. Thanks